test/services/users.test.ts

```diff
@@ -1,13 +1,245 @@
+};
+let userId: NullableId;
+const unexistedId = '783deee0-3732-11ec-9301-13adbda06b66';
 
 describe('\'users\' service', () => {
-  // before
-  // save service globally and use in each test
-  // 
-  it('registered the service', () => {
-    const service = app.service('users');
+  it('It should succeed - registered the service', () => {
+    expect(service).to.exist;
+  });
+
+  describe('create user', () => {
+
+    after(async () => {
+      await service.remove(userId);
+    });
+
+    it('should succeed - creates user', async () => {
+      const response = await service.create(userInfo);
+      const { id, email } = response;
+      userId = id;
+      expect(id).to.exist;
+      expect(email).to.equal(userInfo.email);
+      expect(response).to.have.keys(['id', 'firstName', 'lastName', 'email', 'createdAt', 'updatedAt', 'password', 'title']);
+    });
+
+    it('should fail - email already in use', async () => {
+      try{
+        await service.create(userInfo);
+        expect.fail('call should have failed');
+      } catch (err: any) {
+        const {code, message} = err;
+        expect(code).to.equal(400);
+        expect(message).to.equal('Email is already in use. Please log in');
+      }
+    });
+
+    it('should fail - missed data', async () => {
+      try{
+        await service.create(missedUserData);
+        expect.fail('call should have failed');
+      } catch (err: any) {
+        const {code, message} = err;
+        expect(code).to.equal(400);
+        expect(message).to.equal('Invalid data');
+      }
+    });
+
+  });
+
+  describe('get user', () => {
+
+    before(async () => {
+      const {id} = await service.create(userInfo);
+      userId = id;
+    });
+    after(async () => {
+      await service.remove(userId);
+    });
+
+    it('should succeed - get user by id', async () => {
+      const { email, id } = await service.get(userId);
+      expect(id).to.exist;
+      expect(email).to.equal(userInfo.email);
+    });
+
+    it('should fail - not existed userId', async () => {
+      try{
+        await service.get(unexistedId);
+        expect.fail('call should have failed');
+      } catch (err: any) {
+        const {code} = err;
+        expect(code).to.equal(404);
+      }
+    });
+
+    it('should fail - invalid userId', async () => {
+      try{
+        await service.get('123456');
+        expect.fail('call should have failed');
+      } catch (err: any) {
+        const {code, message} = err;
+        expect(code).to.equal(400);
+        expect(message).to.equal('ValidationError: "id" must be a valid GUID');
+      }
+    });
+  });
+
+  describe('find user', () => {
+
+    before(async () => {
+      const {id} = await service.create(userInfo);
+      userId = id;
+    });
+    after(async () => {
+      await service.remove(userId);
+    });
+
+    it('should succeed - find user', async () => {
+      const {total} = await service.find();
+      expect(total).not.to.equal(0);
```

Comment:
Since tests are running on dev env? I don't know how many users are in "total" field
So before test I add one user - so "total" field couldn't be zero
That's why , test is checking if total counter is not equal to zero

Reply:
Ahh, ok. sounds good then.

Also, please check what other fields there, using expect keys.